rasa/nlu/training_data/training_data.py

```diff
@@ -285,13 +295,21 @@ def nlu_as_yaml(self) -> Text:
             RasaYAMLWriter,
         )
 
-        return RasaYAMLWriter().dumps(self)
+        # avoid dumping NLG data (responses). this is a workaround until we
+        # can remove the distinction between nlu & nlg when converting to a string
+        # (so until after we remove markdown support)
+        no_responses_training_data = copy.copy(self)
+        no_responses_training_data.responses = {}
+
+        return RasaYAMLWriter().dumps(no_responses_training_data)
 
     def persist_nlu(self, filename: Text = DEFAULT_TRAINING_DATA_OUTPUT_PATH):
 
-        if filename.endswith("json"):
+        if Path(filename).suffix in JSON_FILE_EXTENSIONS:
             rasa.nlu.utils.write_to_file(filename, self.nlu_as_json(indent=2))
-        elif filename.endswith("md"):
+        elif Path(filename).suffix in MARKDOWN_FILE_EXTENSIONS:
+            rasa.nlu.utils.write_to_file(filename, self.nlu_as_markdown())
+        elif Path(filename).suffix in YAML_FILE_EXTENSIONS:
             rasa.nlu.utils.write_to_file(filename, self.nlu_as_markdown())
```

Comment:
I'd also add a test which checks the correct persisting behavior 
```suggestion
            rasa.nlu.utils.write_to_file(filename, self.nlu_as_yaml())
```